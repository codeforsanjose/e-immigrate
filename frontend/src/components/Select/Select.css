.RequiredError {
    display: none;
    color: var(--color-secondary);
}
