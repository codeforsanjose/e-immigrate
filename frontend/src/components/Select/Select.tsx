import React from 'react';
import './Select.css';
import { ReactSetter } from '../../types/common';
import { CommonComponentProps } from '../../types/CommonComponentProps';
import { AutoRequiredErrorDiv } from '../RequiredErrorPresenter';
import { useQuestionnaireResponseContext } from '../../contexts/QuestionnaireResponseContext';
type SelectProps = CommonComponentProps & {
    selectAnswers?: Array<string>;
    values: Array<string>;
    setShowFollowUp: ReactSetter<Record<string, boolean>>;
};
export function Select(props: SelectProps) {
    const {
        q, 
        selectAnswers, 
        values,
    } = props;
    const { 
        collectAnswer,
        bindField,
        questionnaireResponse,
    } = useQuestionnaireResponseContext();

    const valueForSlug = questionnaireResponse[q.slug];
    const showError = valueForSlug == null || valueForSlug === '';

    React.useEffect(() => {
        if (selectAnswers == null) {
            console.error(`'selectAnswers' was null`);
            return;
        }
        collectAnswer(q.slug, selectAnswers[0]);
    }, [collectAnswer, q.slug, selectAnswers]);
    const onChange = React.useCallback((e: React.ChangeEvent<HTMLSelectElement>) => {
        collectAnswer(q.slug, e.target.value);
    }, [collectAnswer, q.slug]);
    return (
        <>
            <select
                name={q.slug}
                required={q.required}
                {...bindField(q.slug)}
                onChange={onChange}
            >
                {selectAnswers?.map((option, idx) => {
                    return (
                        <option
                            value={values[idx - 1]}
                            key={`${q.slug}-${option}`}
                        >
                            {option}
                        </option>
                    );
                })}
            </select>
            <AutoRequiredErrorDiv show={showError} />
        </>
    );
}
