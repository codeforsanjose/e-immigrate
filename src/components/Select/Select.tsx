import React, { useEffect } from 'react';
import './Select.css';
import { QData, BindFieldFunction, CollectAnswerFunction, ReactSetter } from '../../types/common';
type SelectProps = {
    attributes: {
        q: QData;
        selectAnswers?: Array<string>;
        values: Array<string>;
        bindField: BindFieldFunction;
        collectAnswer: CollectAnswerFunction;
        setShowFollowUp: ReactSetter<Record<string, boolean>>;
        content: {
            errorMessage: string;
        };
    }
};
export function Select(props: SelectProps) {
    const {
        attributes: { 
            q, 
            selectAnswers, 
            bindField, 
            collectAnswer, 
            content, 
            values,
        },
    } = props;
    useEffect(() => {
        if (selectAnswers == null) {
            console.error(`'selectAnswers' was null`)
            return;
        }
        collectAnswer(q.slug, selectAnswers[0]);
    }, []);
    return (
        <>
            <select
                name={q.slug}
                required={q.required}
                {...bindField(q.slug)}
                onChange={(e) => collectAnswer(q.slug, e.target.value)}
            >
                {selectAnswers &&
                    selectAnswers.map((option, idx) => {
                        return (
                            <option
                                value={values[idx - 1]}
                                key={`${q.slug}-${option}`}
                            >
                                {option}
                            </option>
                        );
                    })}
            </select>
            <div className="RequiredError">*{content.errorMessage}</div>
        </>
    );
}

