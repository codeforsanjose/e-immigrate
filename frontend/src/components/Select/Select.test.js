import React from 'react';
import Select from './Select';

import renderer from 'react-test-renderer';

it('Select renders correctly', () => {
    const tree = renderer
        .create(
            <Select
                q={{}}
                answers={[]}
                values={[]}
                bindField={() => {}}
                showFollowUp={{}}
                setShowFollowUp={{}}
                collectAnswer={{}}
                content={{}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
