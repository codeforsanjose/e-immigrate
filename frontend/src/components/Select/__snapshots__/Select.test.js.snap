// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders correctly 1`] = `
Array [
  <select
    onChange={[Function]}
  />,
  <div
    className="RequiredError"
  >
    *
  </div>,
]
`;
